pkg/operation/botanist/secrets.go

```diff
@@ -67,7 +65,12 @@ func (b *Botanist) InitializeSecretsManagement(ctx context.Context) error {
 		b.generateGenericTokenKubeconfig,
 		b.reconcileWildcardIngressCertificate,
 		// TODO(rfranzke): Remove this function in a future release.
-		b.reconcileGenericKubeconfigSecret,
+		func(ctx context.Context) error {
+			return client.IgnoreNotFound(b.SeedClientSet.Client().Delete(ctx, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{
+				Name:      v1beta1constants.SecretNameGenericTokenKubeconfig,
+				Namespace: b.Shoot.SeedNamespace,
+			}}))
+		},
```

Comment:
Why are we deleting this secret here?

Reply:
This is the legacy secret with a static name which was deprecated and should not be used anymore. We need to clean it up here. Since a long while, `gardenlet` generates immutable/unique generic token kubeconfig secrets with a hash suffix.